piet-wgsl/src/shaders/preprocess.rs

```diff
@@ -41,25 +41,41 @@ pub fn preprocess(
 ) -> String {
     let mut output = String::with_capacity(input.len());
     let mut stack = vec![];
-
-    for (line_number, line) in input.lines().enumerate() {
-        let trimmed = line.trim();
-        if trimmed.starts_with("#") {
-            let trimmed = &trimmed[1..];
-            let val_idx = trimmed
+    'outer: for (line_number, mut line) in input.lines().enumerate() {
```

Comment:
Oooh, fancy, labeled loops. I found the flow a little hard to follow, but don't have a suggestion how to improve it.

Reply:
Yeah. In https://github.com/linebender/piet-gpu/pull/196/commits/40f808b366b4a7087b4da97f756e480c5265c859 I tried a different way, but that was really messy.